frontend/src/metabase/query_builder/actions/writeback.ts

```diff
@@ -0,0 +1,42 @@
+import _ from "underscore";
+import { t } from "ttag";
+
+import { addUndo } from "metabase/redux/undo";
+
+import { Dispatch, GetState } from "metabase-types/store";
+
+import { getQuestion } from "../selectors";
+
+import { apiUpdateQuestion } from "./core";
+
+export const turnQuestionIntoAction = () => async (
+  dispatch: Dispatch,
+  getState: GetState,
+) => {
+  const question = getQuestion(getState());
+  const action = question?.setIsAction(true);
```

Comment:
It shouldn't actually be nullish in any case if this function is called. The reason is that `question` in QB state can be `null` for a short period of time during QB init. Because of that, `question` is typed as `Question | null` and we have do add null checks here and there.